llama_index/readers/discord_reader.py

```diff
@@ -17,7 +17,7 @@
 
 
 async def read_channel(
-    discord_token: str, channel_id: int, limit: Optional[int], oldest_first: bool
+    discord_token: str, channel_id: int, limit: Optional[int], oldest_first: bool, complete_metadata: bool = False,
```

Comment:
hmmm, the name `complete_metadata` seems to conflict with other concepts in llama-index (i.e. metadata for document objects)

Maybe a name like `full_message_data` makes more sense?

Reply:
I wonder if it would be prudent to make the full data consumed by default, and then be able to specify type-filters/flags to include specified information from the message data?

e.g.
```
USERNAME_FLAG = 0b0001
CONTENT_FLAG = 0b0010
# etc..

documents = DiscordReader(discord_token=discord_token).load_data(
    channel_ids = channel_ids
    flags = USERNAME_FLAG | CONTENT_FLAG
)
```

Reply:
to be honest, it might be ideal if this information made it's way to the metadata of each document, but this would require a bit more work

```
Document(text=text, metadata={'author': ..., ....})
```

Reply:
@logan-markewich see what you think of the approach in the [latest commit](https://github.com/jerryjliu/llama_index/pull/7906/commits/f5bb0ca77909a10f7029d0b59c969e5e16c44c85). We opted for the return to be a mapping such that each single message is wrapped in a Document containing the content in addition to the metadata by property assignment. 

Notes: 
* Querying the engine obtained similar results as described in the initial PR. 
* Also, we view this is a starting point. If other use cases require additional metadata, it can be extended to support that.